lib/parsers/get-package-lock-dep-graph.ts

```diff
@@ -0,0 +1,167 @@
+import * as _isEmpty from 'lodash.isempty';
+import {
+  Dep,
+  getTopLevelDeps,
+  Lockfile,
+  LockfileType,
+  ManifestFile,
+  Scope,
+} from './index';
+import { InvalidUserInputError, OutOfSyncError } from '../errors';
+import { DepGraph, DepGraphBuilder } from '@snyk/dep-graph';
+import {
+  PackageLock,
+  PackageLockDep,
+  PackageLockDeps,
+} from './package-lock-parser';
+import { eventLoopSpinner } from 'event-loop-spinner';
+
+export async function getPackageLockDepGraph(
+  manifestFile: ManifestFile,
+  lockfile: Lockfile,
+  includeDev = false,
+  strict = true,
+): Promise<DepGraph> {
+  if (lockfile.type !== LockfileType.npm) {
+    throw new InvalidUserInputError(
+      'Unsupported lockfile provided. Please ' + 'provide `package-lock.json`.',
+    );
+  }
+  const packageLock = lockfile as PackageLock;
+
+  recursiveLinkRequire(packageLock.dependencies || {}, [packageLock]);
+
+  const rootPkgInfo = {
+    name: manifestFile.name,
+    version: manifestFile.version,
+  };
+
+  const pkgManager = {
+    name: LockfileType.npm,
+  };
+
+  const depGraphBuilder = new DepGraphBuilder(pkgManager, rootPkgInfo);
+
+  // const nodeVersion = manifestFile?.engines?.node;
+  // if (nodeVersion) {
+  //   _set(depTree, 'meta.nodeVersion', nodeVersion);
+  // }
+
+  // asked to process empty deps
+  if (_isEmpty(manifestFile.dependencies) && !includeDev) {
+    return depGraphBuilder.build();
+  }
+
+  const topLevelDeps: Dep[] = getTopLevelDeps(manifestFile, includeDev);
+
+  const devMemoizationMap: Map<PackageLockDep, NodeId> = new Map();
+  const prodMemoizationMap: Map<PackageLockDep, NodeId> = new Map();
+
+  for (const dep of topLevelDeps) {
+    const dependency = packageLock.dependencies?.[dep.name];
+    if (/^file:/.test(dep.version)) {
```

Comment:
This is a bit confusing and can make one thing (like myself) that we may see an infinite loop here.

Since we're already iterating top-level-deps, what is the meaning of adding another top-level dep?

Reply:
`addTopLevelDep` name is responsible for the confusion, so I renamed it to `addTopLevelDepNodeToGraph`